app.go

```diff
@@ -208,6 +221,102 @@ func (h *Headscale) watchForKVUpdatesWorker() {
 	// more functions will come here in the future
 }
 
+func (h *Headscale) grpcAuthenticationInterceptor(ctx context.Context,
+	req interface{},
+	info *grpc.UnaryServerInfo,
+	handler grpc.UnaryHandler) (interface{}, error) {
+
+	// Check if the request is coming from the on-server client.
+	// This is not secure, but it is to maintain maintainability
+	// with the "legacy" database-based client
+	// It is also neede for grpc-gateway to be able to connect to
+	// the server
+	p, _ := peer.FromContext(ctx)
+
+	if IsLocalhost(p.Addr.String()) {
```

Comment:
Would this be a problem with a reverse proxy? It could be shown as coming from localhost...

Reply:
Good point, I will make a config option to turn that off. 

That would be a problem while the api key part isn't ready tho...

Reply:
This feels like a use case for unix sockets and then optionally tcp (with some bearer token or mutual tls)

Reply:
its done